riscv-asm.md

```diff
@@ -320,21 +295,20 @@ Load Immediate
 The following example shows the `li` pseudo instruction which
 is used to load immediate values:
 
-```
-.section .text
-.globl _start
-_start:
-
-.equ CONSTANT, 0xdeadbeef
+```assembly
+	.equ	CONSTANT, 0xdeadbeef
 
-        li a0, CONSTANT
+	li	a0, CONSTANT
 ```
 
-which, for RV32I, generates the following assembler output, as seen by objdump:
+Which, for RV32I, generates the following assembler output, as seen by `objdump`:
 
-```
-   0:   deadc537                lui     a0,0xdeadc
-   4:   eef50513                addi    a0,a0,-273
+```assembly
+0000000000000000 <.text>:
+   0:	00038537          	lui	a0,0x38
+   4:	ab75051b          	addiw	a0,a0,-1353
+   8:	00e51513          	slli	a0,a0,0xe
+   c:	eef50513          	addi	a0,a0,-273 # 37eef <CONSTANT-0xdeaa4000>
```

Comment:
This is the rv64 sequence for a non-sign extended 32-bit constant (without the B extension).  The comment above says rv32i.  For rv64i, the constant should be 0xffffffffdeadbeef and you get the same two instructions as rv32i.